### DISCUSSION TITLE
How to create a branch in octokit

### DISCUSSION THREAD

**Discussion by blackgirlbytes** [octokit/octokit.js#2198] [Q&A]
I have a question about creating a branch with octokit. I found some documentation, and I tried doing the below:
```
octokit.rest.git.createRef({
  owner,
  repo,
  ref,
  sha,
});
```
 but when I do that the error i get back is that rest is undefined, which like makes sense i guess cuz rest is not a property on ocktokit.
Are there any suggestions on how to create a branch with octokit?
My ultimate goal is to:
- create a branch 
- commit to that branch
- and open a pull request

**Comment by Vanals** [comment]
Hi @blackgirlbytes ! I am having an issue related to creating a branch I wonder if u have found a solution and could help me on the following? Thanks
https://github.com/octokit/octokit.js/discussions/2284

**Comment by oscard0m** [comment]
Hi @blackgirlbytes,

Sorry for the late answer to your question. 

Could you share the missing lines of code where you import `octokit` and instantiate it? Your problem is probably related to that.
